src/core/jit/common/json_placeholder.rs

```diff
@@ -82,6 +84,10 @@ impl JsonPlaceholder {
             store
         });
 
+        let input_resolver = InputResolver::new(plan);
+
+        let plan = input_resolver.resolve_input(&Variables::default()).unwrap();
+
         Synth::new(plan, store)
```

Comment:
May be `ConstSynth` could perform this transformation of the plan using `plan.try_map` while synthesizing?